api/client/sg.cc

```diff
@@ -422,20 +422,12 @@ static void sub_sg_rule_del_help(void) {
     global_parameter_help();
 }
 
-static int sub_sg_rule_del(int argc, char **argv,
-                            const GlobalOptions &options) {
-    if (argc >= 5 && string(argv[4]) == "help") {
-        sub_sg_rule_add_help();
-        return 0;
-    }
-
-    if (argc <= 5) {
-        sub_sg_rule_del_help();
-        return 1;
-    }
+static bool check_hash(const string &param) {
+    return param.find("-") == 0;
```

Comment:
s/check_hash/is_hash/
you use "c" so a char *, but find can take a char in parameter, so you could simply use 'c', a char
`return param.find('-') ==  string::npos`

http://www.cplusplus.com/reference/string/string/find/